board/app/models/favorite.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+class Favorite extends AppModel 
+{
+
+    public static function countFavoriteByCommentId($comment_id)
+    {
+        $db = DB::conn();
+        $total_favorite = $db->value('SELECT COUNT(*) FROM favorite WHERE comment_id =?', array($comment_id));
+    
+        return $total_favorite;
+    }
+
+    public static function isCommentFavorited($comment_id, $user_id)
+    {
+        $db = DB::conn();
+        $comment_favorited = $db->rows('SELECT * FROM favorite WHERE comment_id = ? AND user_id = ?', array($comment_id, $_SESSION['user_id']));
+        
+        return !$comment_favorited;
+    }
+
+    public static function countFavoriteByUserId($user_id)
+    {
+        $db = DB::conn();
+        $fave = $db->value('SELECT COUNT(*) FROM favorite WHERE user_id = ?', array($user_id));
+        
+        return $fave;
+    }
+
+    public static function getAllFavorites($offset, $limit, $user_id)
+    {
+        $favorites = array();
+        $db = DB::conn();                  
+        $rows = $db->rows("SELECT * FROM favorite WHERE user_id = ? LIMIT {$offset}, {$limit}", array($user_id));
```

Comment:
Done po.
I throwed exceptions for consistency po since it has been already applied to other parts of my code.
Thank you :bow:
--> https://github.com/maragatchalian/FoodFacts/commit/a2b7297fb42d73c941b8acc623787aadc730de45